### ISSUE TITLE
Fix number inconsistencies due to conversion from uint64 to float64 to int64 and back to uint64.

### ISSUE THREAD

**Issue by lpoli** [0chain/0chain#1452]
When `encoding/json` unmarshalls into `map[string]interface{}` it type casts all the numbers to float64. Type casting to float64 will cause rounding errors i.e. it rounds off to close number that it can represent with float64. https://go.dev/play/p/wpZgX4fOU59

In https://github.com/0chain/0chain/blob/staging/code/go/0chain.net/smartcontract/dbs/types.go#L7, the type of `Updates` field is `map[string]interface{}` so when unmarshalling fields such as `Balance` it will type cast to float64.

Error in system-test:
```
./zbox upload --localpath ~/Downloads/ImageOptim.tbz2 --allocation $alloc --remotepath /ImageOptim1.tbz2                                                                                                     11:09:38
0chain-core-sdk [INFO]   2022/06/29 11:16:48.876529 wallet.go:292: ******* Wallet SDK Version:v1.7.2-2-g3237f5c-dirty ******* (SetLogFile)
0box-sdk        [INFO]   2022/06/29 11:16:48.879494 sdk.go:73: ******* Storage SDK Version: v1.7.2-2-g3237f5c-dirty *******
0chain-core-sdk [INFO]   2022/06/29 11:16:51.159487 wallet.go:415: ******* Wallet SDK Version:v1.7.2-2-g3237f5c-dirty ******* (InitZCNSDK)
0box-sdk        [DEBUG]  2022/06/29 11:16:52.767852 networkworker.go:105: Get network result:{"miners":["https://test.0chain.net/miner03","https://test.0chain.net/miner02","https://test.0chain.net/miner01"],"sharders":["https://test.0chain.net/sharder01","https://test.0chain.net/sharder02"]}
Error fetching the allocation. allocation_fetch_error: Error fetching the allocation.internal_error: can't get allocation: error retrieving allocation: 8a274a48f4ff78adb24b8fa9a074832926ea33d20f941f37772655d46ea9fbda, error: sql: Scan error on column index 16, name "read_price_max": converting driver.Value type int64 ("-9223372036854775808") to a uint64: invalid syntax

```

For tests with gorm please check https://go.dev/play/p/RVxnninRu1p.

**Comment by cnlangzi** [comment]
it breaks multiple things
```
2022-07-01T16:34:38.039+0800	ERROR	challenge/protocol.go:51	Error verifying the challenge response transaction	{"err:": "transaction_verify_error: Error unmarshaling verify output: \"challenge_reward_error: can't move tokens to blobber: uint64 overflows int64\" json: cannot unmarshal string into Go value of type map[string]json.RawMessage", "txn": "79720e773483fba0726fca3940ee89c472cdfd2f6a79eba2a079f6735383bdd7", "challenge_id": "0fb2388f3cf94f6127140b30bf9b3bff16d3e9126a0dfd01afc0ea151a70175c"}
github.com/0chain/blobber/code/go/0chain.net/blobbercore/challenge.(*ChallengeEntity).SubmitChallengeToBC
	/Users/geax/code/geax/github.com/0chain/blobber/code/go/0chain.net/blobbercore/challenge/protocol.go:51
github.com/0chain/blobber/code/go/0chain.net/blobbercore/challenge.(*ChallengeEntity).CommitChallenge
	/Users/geax/code/geax/github.com/0chain/blobber/code/go/0chain.net/blobbercore/challenge/protocol.go:279
github.com/0chain/blobber/code/go/0chain.net/blobbercore/challenge.commitChallenge
	/Users/geax/code/geax/github.com/0chain/blobber/code/go/0chain.net/blobbercore/challenge/challenge.go:195
github.com/0chain/blobber/code/go/0chain.net/blobbercore/challenge.commitProcessed.func2
	/Users/geax/code/geax/github.com/0chain/blobber/code/go/0chain.net/blobbercore/challenge/challenge.go:177
2022-07-01T16:34:38.056+0800	ERROR	challenge/protocol.go:287	[challenge]submit: Error while submitting challenge to BC.	{"challenge_id": "0fb2388f3cf94f6127140b30bf9b3bff16d3e9126a0dfd01afc0ea151a70175c", "error": "transaction_verify_error: Error unmarshaling verify output: \"challenge_reward_error: can't move tokens to blobber: uint64 overflows int64\" json: cannot unmarshal string into Go value of type map[string]json.RawMessage"}
github.com/0chain/blobber/code/go/0chain.net/blobbercore/challenge.(*ChallengeEntity).CommitChallenge
	/Users/geax/code/geax/github.com/0chain/blobber/code/go/0chain.net/blobbercore/challenge/protocol.go:287
github.com/0chain/blobber/code/go/0chain.net/blobbercore/challenge.commitChallenge
	/Users/geax/code/geax/github.com/0chain/blobber/code/go/0chain.net/blobbercore/challenge/challenge.go:195
github.com/0chain/blobber/code/go/0chain.net/blobbercore/challenge.commitProcessed.func2
	/Users/geax/code/geax/github.com/0chain/blobber/code/go/0chain.net/blobbercore/challenge/challenge.go:177
```

**Comment by cnlangzi** [comment]
if it is json issue, it will be fixed by https://github.com/0chain/0chain/pull/1448

**Comment by cnlangzi** [comment]
fixed by #1448
